Makefile

```diff
@@ -151,7 +151,7 @@ vet: ## Run go vet against code.
 	go vet ./...
 
 .PHONY: test
-test: manifests generate generate-protobuf fmt vet envtest ## Run tests.
+test: manifests generate docker-generate-protobuf fmt vet envtest ## Run tests.
```

Comment:
@Rakshith-R what is the use of this change?

Reply:
> @Rakshith-R what is the use of this change?

It avoids running `generate-protobuf` on the user's system.
`docker-generate-protobuf` takes care of all the dependencies.

`docker-build` calls `test` which in turn calls `generate-protobuf` which is very inconvenient.

Reply:
we might get in to the problem when there is a protobuf version update, Currently we are seeing the difference in the CI and pushing the changes as extra commit to keep the consistent pb files for the internal protobuf, if we are planning to change it we need to avoid having this problem please make required changes to `github actions` as well.

Reply:
i looked at the Makefile , the above is not required. please take care of fixing the CI problem

Reply:
i think problem is docker build running test, why do we need to run test why building the container it should only do the build not test? can you adjust generating the deps when running docker-build